### PR TITLE
Cypress: PSoC-6: Add SCB[spi] driver

### PR DESCRIPTION
**Pull request by nandojve**
Add master spi driver for PSoC-6. Initial config is added for `cy8ckit_062_ble` board.

### CODE REVIEW

**Comment by nandojve** [comment]
buildkite errors are non related.

**Comment by nandojve** [comment]
- Rebase
- Update west.yml with new hash value
- Rework `device_pm_control_nop` in favor of `NULL`

File: drivers/spi/spi_psoc6.c [hunk 1]
```diff
@@ -0,0 +1,419 @@
+/*
+ * Copyright (c) 2021 ATL Electronics.
```
Comment (MaureenHelm) [inline drivers/spi/spi_psoc6.c:2]: This file appears to be based on `drivers/spi/spi_mcux_dspi.c`. Please carry over the copyright

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/spi/spi_psoc6.c (hunk 1 above)
Comment (nandojve) [reply drivers/spi/spi_psoc6.c]: Indeed, some ideas came from `spi_mcux_dspi.c`. Added credits as requested.

**Review by MaureenHelm — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**